### PR TITLE
docker(-compose) should be used to make things easy

### PR DESCRIPTION
**Pull request by etherneet**
Now you just have to `docker-compose up -d` and it works without contaminating your whole system.

It also doesn't need the `/var/run/docker.sock` inside the container.

glhf

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,17 +16,30 @@ RUN apt-get update && \
...
 
 ADD . /app
 
 WORKDIR /app/embark
 
 ADD embark/requirements.txt /app/embark/requirements.txt
 
-RUN yes | sudo /app/emba/installer.sh -D  && \
-    sudo pip3 install uwsgi -I --no-cache-dir && \
+RUN yes | sudo pip3 install uwsgi -I --no-cache-dir && \
```
Comment (m-1-k-3) [inline Dockerfile:27]: With this change we have no EMBA installation within the EMBArk container anymore. This is needed as scanning backend. Do I miss something?

File: Dockerfile [hunk 2]
```diff
@@ -16,17 +16,30 @@ RUN apt-get update && \
...
+  mkdir -p /app/embark/static/external/css && \
+  wget -O /app/embark/static/external/scripts/jquery.js https://code.jquery.com/jquery-3.6.0.min.js && \
+  wget -O /app/embark/static/external/scripts/confirm.js https://cdnjs.cloudflare.com/ajax/libs/jquery-confirm/3.3.2/jquery-confirm.min.js && \
+  wget -O /app/embark/static/external/scripts/bootstrap.js https://cdn.jsdelivr.net/npm/bootstrap@5.1.1/dist/js/bootstrap.bundle.min.js && \
+  wget -O /app/embark/static/external/scripts/datatable.js https://cdn.datatables.net/v/bs5/dt-1.11.2/datatables.min.js && \
+  wget -O /app/embark/static/external/scripts/charts.js https://cdn.jsdelivr.net/npm/chart.js@3.5.1/dist/chart.min.js && \
+  wget -O /app/embark/static/external/css/confirm.css https://cdnjs.cloudflare.com/ajax/libs/jquery-confirm/3.3.2/jquery-confirm.min.css && \
+  wget -O /app/embark/static/external/css/bootstrap.css https://cdn.jsdelivr.net/npm/bootstrap@5.1.1/dist/css/bootstrap.min.css && \
+  wget -O /app/embark/static/external/css/datatable.css https://cdn.datatables.net/v/bs5/dt-1.11.2/datatables.min.css && \
+  find /app/embark/static/external/ -type f -exec sed -i '/sourceMappingURL/d' {} \;
```
Comment (BenediktMKuehne) [inline Dockerfile:40]: If we move these lines into the Dockerfile , the corresponding lines in the installer-script have to be deleted

File: Dockerfile (hunk 1 above)
Comment (BenediktMKuehne) [reply Dockerfile:27]: Right now embark is running the emba-analysis in Developer mode.

image

File: template.env [hunk 3]
```diff
@@ -1,8 +1,8 @@
 DATABASE_NAME=
 DATABASE_USER=
 DATABASE_PASSWORD=
-DATABASE_HOST=
-DATABASE_PORT=
+DATABASE_HOST=mysql
```
Comment (BenediktMKuehne) [inline template.env:4]: This is the template file, the default settings are set in installer.sh line 116-

File: Dockerfile (hunk 1 above)
Comment (m-1-k-3) [reply Dockerfile:27]: EMBA is running within the docker container of EMBArk in dev mode. With this PR the installation process of EMBA will be removed and will break the complete system.

File: Dockerfile (hunk 2 above)
Comment (m-1-k-3) [reply Dockerfile:40]: as we need the intaller script I prefer to leave this in the installer script

**Comment by m-1-k-3** [comment]
From my perspective this PR will need a bunch of other things to change. Currently we should leave the job in the installer. Probably we can pick up some of the mentioned changes later.

**Pull request closed without merging**